client/src/handlers/handlers.ts

```diff
@@ -22,138 +18,45 @@ export function makeHandleRunJob(client: LanguageClient, request: jobs.Request)
-    }
-  })
-
   return missing
 }
 
 /**
- * Launch the REPL in the given terminal.
- */
-async function launchReplInTerminal(
-  terminal: vscode.Terminal,
-  context: vscode.ExtensionContext,
-  launchOptions: LaunchOptions,
-) {
-  const cmd = await getJVMCmd(context, launchOptions)
-
-  cmd.push('repl')
-  if (vscode.workspace.getConfiguration('flix').get('explain.enabled')) {
-    cmd.push('--explain')
-  }
-  cmd.push(...getExtraFlixArgs())
-
-  terminal.sendText(quote(cmd))
-}
-
-/**
- * returns an active terminal with prefix name `flix`.
- *
- * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix`.
- *
- * @return vscode.Terminal
- */
-function getFlixTerminal() {
-  const activeTerminals = vscode.window.terminals
-  for (const element of activeTerminals) {
-    if (element.name.substring(0, 4) === `flix`) {
-      return element
-    }
-  }
-  const terminal = vscode.window.createTerminal(`flix-` + countTerminals.toString())
-  countTerminals += 1 //creating a new terminal since no active flix terminals available.
-  return terminal
-}
-
-/**
- * returns an new active terminal with prefix name `flix`.
- *
- * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix` and returns it.
- *
- * If there are already `n` active terminals exist with prefix name `flix`, it creates a new terminal with name `flix n+1`
- *
- * @return vscode.Terminal
- */
-function newFlixTerminal() {
-  const terminal = vscode.window.createTerminal(`flix-` + countTerminals.toString())
-  countTerminals += 1
-  return terminal
-}
-
-/**
- * an array of string arguments entered by user in flix extension settings `Extra Flix Args`.
- * @returns string[]
- */
-
-function getExtraFlixArgs() {
-  const arg: string = vscode.workspace.getConfiguration('flix').get('extraFlixArgs')
-  return arg.split(' ')
-}
-/**
- * takes a string and a terminal and passes that string to the terminal.
- *
- * @param cmd string (a terminal command) to pass to the terminal.
- *
- * @param terminal vscode.Terminal
- *
- * @return void
- */
-function passCommandToTerminal(cmd: string[], terminal: vscode.Terminal) {
-  terminal.show()
-  terminal.sendText(quote(cmd))
-}
-
-/**
- * Opens an input box to ask the user for input.
- * uses the `vscode.window.showInputBox` function with custom `prompt` and `placeHolder` and value of `ignoreFocusOut` to be `true`.
- *
- *
- * @return A promise that resolves to a string the user provided or to `undefined` in case of dismissal.
+ * Launches a new REPL as a shell process.
  */
-async function takeInputFromUser() {
-  const { prompt, placeHolder } = USER_MESSAGE.ASK_PROGRAM_ARGS()
-  const input = await vscode.window.showInputBox({
-    prompt,
-    placeHolder,
-    ignoreFocusOut: true,
-  })
-  return input
+async function launchRepl(context: vscode.ExtensionContext, launchOptions: LaunchOptions) {
+  const { cmd, args } = await getJVMCmd(context, launchOptions)
```

Comment:
getJVMCmd will pass extra args, yes?

Reply:
Only JVM args, but it should be fixed now